### ISSUE TITLE
Conflict with loading org.ec4j.core

### ISSUE THREAD

**Issue by paul-dingemans** [nbadal/ktlint-intellij-plugin#598]
In #575 it is remarked that after upgrade to version `0.25.0` that the problem with reading `.editorconfig` properties without values is not resolved:
> I upgraded the plugin but I still get 
> Caused by: java.lang.AssertionError: An error occurred while reading the '.editorconfig':
> path:/Sourcecode/project/.editorconfig:17:19: Property 'ij_visual_guides' has no value
>
> That the drive letter is missing in the path is also a sign still the old ktlint is used, as this was fixed.
:-(

_Originally posted by @Vampire in https://github.com/nbadal/ktlint-intellij-plugin/issues/575#issuecomment-2486761578_

This problem is caused by multiple versions of the `org.ec4j.core` library being loaded via the different ktlint ruleset versions.

**Comment by paul-dingemans** [comment]
`libs.versions.toml`:
```
ktlint = "1.4.1"

ktlintRuleEngine = {  group = "com.pinterest.ktlint", name = "ktlint-rule-engine", version.ref = "ktlint" }
ktlintCliRulesetCore = {  group = "com.pinterest.ktlint", name = "ktlint-cli-ruleset-core", version.ref = "ktlint" }
ktlintCliReporterCore = {  group = "com.pinterest.ktlint", name = "ktlint-cli-reporter-core", version.ref = "ktlint" }
ktlintCliReporterBaselineCore = {  group = "com.pinterest.ktlint", name = "ktlint-cli-reporter-baseline", version.ref = "ktlint" }
ktlintRulesetStandard = {  group = "com.pinterest.ktlint", name = "ktlint-ruleset-standard", version.ref = "ktlint" }
```
In `libs.version.toml` of `ktlint` version `1.4.1`:
```
ec4j = "org.ec4j.core:ec4j-core:1.1.0"
```
The `ec4j` dependency in `ktlint` is exposed via `ktlint-rule-engine` and `ktlint-rule-engine-core` dependencies.

In the plugin in module `ktlint-lib` multiple versions of the `ktlint-ruleset-standard` are being included. Each of those versions has a direct dependency on `ktlint-rule-engine-core`, and a transitive dependency on that same artifact via the intermediary `ktlint-cli-ruleset-core`.

In the `build.gradle.kts` of a ruleset in `ktlint-lib`, I was not able to fully exclude the dependency on `org.ec4j.core`. But I am able to relocate it, as is visble below:
![Screenshot 2024-12-09 at 17 42 39](https://github.com/user-attachments/assets/ab29cae7-bad7-4106-b263-897e7a4cc605)

When merging the different ruleset versions into `ktlint-lib` the result is as shown below:
![Screenshot 2024-12-09 at 17 43 51](https://github.com/user-attachments/assets/ce163d02-bc09-454d-baa0-927c5bf89b7a)

Although this looks good, as we have multiple versions of the `org.ec4j.core` library, the problem is that they both seem to be the `0.3.0` version.

**Comment by paul-dingemans** [comment]
With much appreciated help of @Vampire the problem resolve by fixing the dependencies in the ruleset (for example in `1.3.1` version) blocks as follows:
```
dependencies {
    implementation("com.pinterest.ktlint:ktlint-ruleset-standard:1.3.1")

    constraints {
        // ec4-core version 0.3.0 which is included in ktlint 1.3.1 fails on '.editorconfig' properties without value
        implementation("org.ec4j.core:ec4j-core:1.1.0") {
            because("Allows '.editorconfig' properties to be defined without any value")
        }
    }
}
```

**Comment by paul-dingemans** [comment]
Resolved by #603
